client/src/App.js

```diff
@@ -13,32 +15,77 @@ class App extends Component {
 	constructor() {
 		super()
 
-		this.updateState = this.updateState.bind(this)
+		this.addList = this.addList.bind(this)
+		this.addListTask = this.addListTask.bind(this)
 		this.state = {
-			lists: JSON.parse(localStorage.getItem('lists')) || listsState
+			lists: JSON.parse(localStorage.getItem('lists')) || initialLists,
+			tasks: JSON.parse(localStorage.getItem('tasks')) || initialTasks
 		};
 	}
 
-	updateState(e) {
-		console.log(e);
-		this.setState(e);
-		localStorage.setItem('lists', JSON.stringify(e));
+	addList(listItem) {
+		const oldLists = this.state.lists
+		const newID = Object.keys(oldLists).length + 1;
+
+		const lists = {
+			...oldLists,
+			[newID]: {
+				title: listItem,
+				tasks: []
+			}
+		}
+
+		this.setState({
+			lists
+		});
+		localStorage.setItem('lists', JSON.stringify(lists));
+	}
+
+	addListTask(taskId, newTask) {
+		const oldTasks = this.state.tasks
+		const randomTaskId = Object.keys(oldTasks).length + 1;
+
+		const tasks = {
+			...this.state.tasks,
+			[randomTaskId]: {
+				text: newTask
+			}
+		}
+
+		const lists = {
+			...this.state.lists,
+			[taskId]: {
+				...this.state.lists[taskId],
+				tasks: [
+					...this.state.lists[taskId].tasks,
+					randomTaskId
+				]
```

Comment:
This **might** become easier to manage if you used a library to help. I have heard that https://immerjs.github.io/immer/docs/introduction can help